### ISSUE TITLE
implicit numeric widening error

### ISSUE THREAD

**Issue by RadoBuransky** [scalatest/scalatest#445]
I am getting "implicit numeric widening" compilation error on the following line:
```
assert(List(1).length == 1)
```
It's not a blocking issue for me, a workaround is easy. This helps even though it doesn't make much sense because on both sides of the equation are integers already:
```
assert(List(1).length.toInt == 1)
```

**Comment by bvenners** [comment]
I am unable to reproduce this. Can you still reproduce it on the latest ScalaTest? If so, can you give more detail on how to reproduce it?

**Comment by RadoBuransky** [comment]
Of course my mistake, I haven't provided important info about Scala compiler options. You have to turn on warnings for numeric widening and then make warnings fatal. Here the problem is that I don't see the actual widening. Is 1 widened from byte to int or what?

    import org.scalatest._
    
    /**
     * ScalaTest 3.0.0-SNAP4 for Scala 2.11
     * Scala 2.11.6 compiler options:
     *   -Ywarn-numeric-widen
     *   -Xfatal-warnings
     */
    class ImplicitNumericWideningSpec extends FlatSpec with Matchers {
      it should "not compile" in {
        assert(List(1).length == 1)
      }
    
      it should "compile" in {
        assert(List(1).length.toInt == 1)
      }
    }

**Comment by retronym** [comment]
Just saw this reported in ensime-server.

I debugged, and the tree being typechecked is:

```
{
  val $org_scalatest_assert_macro_expr = {
    val $org_scalatest_assert_macro_left = result.completions;
    val $org_scalatest_assert_macro_right = 1;
    _root_.org.scalactic.Bool.lengthSizeMacroBool($org_scalatest_assert_macro_left, "length", $org_scalatest_assert_macro_left.length, $org_scalatest_assert_macro_right)
  };
  assertionsHelper.macroAssert($org_scalatest_assert_macro_expr, "")
}
```


From this point:

````
	  at scala.reflect.macros.contexts.Context.typecheck(Context.scala:6)
	  at scala.reflect.macros.Typers$class.typeCheck(Typers.scala:58)
	  at scala.reflect.macros.contexts.Context.typeCheck(Context.scala:6)
	  at org.scalactic.MacroOwnerRepair.repairOwners(MacroOwnerRepair.scala:34)
	  at org.scalactic.BooleanMacro.genMacro(BooleanMacro.scala:784)
	  at org.scalatest.AssertionsMacro$.assert(AssertionsMacro.scala:34)
	  at sun.reflect.GeneratedMethodAccessor1.invoke(Unknown Source:-1)
	  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	  at java.lang.reflect.Method.invoke(Method.java:497)
	  at scala.reflect.macros.runtime.JavaReflectionRuntimes$JavaReflectionResolvers$$anonfun$resolveJavaReflectionRuntime$2.apply(JavaReflectionRuntimes.scala:34)
	  at scala.reflect.macros.runtime.JavaReflectionRuntimes$JavaReflectionResolvers$$anonfun$resolveJavaReflectionRuntime$2.apply(JavaReflectionRuntimes.scala:22)
	  at scala.tools.nsc.typechecker.Macros$class.macroExpandWithRuntime(Macros.scala:756)
```

And the tree that incurs the warning is unpositioned, which results in the compiler emitting the warning without a file/line number.

You could change:

```
class MacroOwnerRepair {
 ...
 val typed = c.typeCheck(expr.tree).asInstanceOf[symtab.Tree]
```

To:

```
 val typed = c.typeCheck(atPos(macroApplication.pos)(expr.tree)).asInstanceOf[symtab.Tree]
```

To at least get a positioned error.

You could then either provide an `Int` overload of `lengthSizeMacroBool`, or synthesize explicit `toLong` calls rather than relying on the implicit numeric widening.

**Comment by cheeseng** [comment]
@RadoBuransky @retronym I submitted a PR here: 

https://github.com/scalatest/scalatest/pull/671

Thanks @retronym for the fix!  I tested and it worked like a charm.

**Comment by bvenners** [comment]
This was fixed 5 months ago in 3.0.x and included in all subsequent milestone releases. Closing now.
